### ISSUE TITLE
SequentialChain throws error finding variable already successfully used

### ISSUE THREAD

**Issue by nigel-daniels** [hwchase17/langchainjs#1578]
I have been following the DeepLearning.AI introduction but have been using the JS version of langchain. In attempting to use the SequentialChain I cannot get the chain to work as it is failing to reuse previously input variables, yet the Python version does. Here is the code I have been attempting:

```
import { ChatOpenAI } from 'langchain/chat_models/openai';
import { ChatPromptTemplate, HumanMessagePromptTemplate } from 'langchain/prompts';
import { LLMChain, SequentialChain } from 'langchain/chains';

// Note: here we use a higher temp for more creativity
let llm = new ChatOpenAI({temperature: 0.9});

let template4 = 'Translate the following review to english:\n\n{review}';
let promptTemplate4 = ChatPromptTemplate.fromPromptMessages([
    HumanMessagePromptTemplate.fromTemplate(template4)
]);
let llmChain4 = new LLMChain({llm: llm, prompt: promptTemplate4, outputKey: 'english_review'});

let template5 = 'Can you summarize the following review in 1 sentence:\n\n{english_review}';
let promptTemplate5 = ChatPromptTemplate.fromPromptMessages([
    HumanMessagePromptTemplate.fromTemplate(template5)
]);
let llmChain5 = new LLMChain({llm: llm, prompt: promptTemplate5, outputKey: 'summary'});

let template6 = 'What language is the following review:\n\n{review}';
let promptTemplate6 = ChatPromptTemplate.fromPromptMessages([
    HumanMessagePromptTemplate.fromTemplate(template6)
]);
let llmChain6 = new LLMChain({llm: llm, prompt: promptTemplate6, outputKey: 'language'});

let template7 = 'Write a follow up response to the following summary in the ' +
    'specified language:\n\nSummary: {summary}\n\nLanguage: {language}';
let promptTemplate7 = ChatPromptTemplate.fromPromptMessages([
    HumanMessagePromptTemplate.fromTemplate(template7)
]);
let llmChain7 = new LLMChain({llm: llm, prompt: promptTemplate7, outputKey: 'followup_message'});

let sequentialChain = new SequentialChain({
    chains: [llmChain4, llmChain5, llmChain6, llmChain7],
    inputVariables: ['review'],
    outputVariables:['english_review', 'summary', 'followup_message'],
    verbose: true
});

let review1 = 'Je trouve le goût médiocre. La mousse ne tient pas, c\'est bizarre. J\'achète les mêmes dans le commerce et le goût est bien meilleur... Vieux lot ou contrefaçon !?';
let response3 = await sequentialChain(review1);
console.log(response3);
```

This fails when attempting to run `llmChain6` and throws the error:

```
[chain/error] [1:chain:sequential_chain > 6:chain:llm_chain] [1ms] Chain run errored with error: "Missing value for input variable \`review\`"
[chain/error] [1:chain:sequential_chain] [6.67s] Chain run errored with error: "Missing value for input variable \`review\`"
```

However `llmChain4` has already used the same variable?

**Comment by borel** [comment]
Hey @nigel-daniels , 
I see that you are using ChatPromptTemplate in classical LLMChain.

```javascript
let promptTemplate6 = ChatPromptTemplate.fromPromptMessages([
    HumanMessagePromptTemplate.fromTemplate(template6)
]);
``` 

and not "classic" prompt like
```javascript
  const template2 = "{input2}";
  const prompt2 = new PromptTemplate({
    template: template2,
    inputVariables: ["input2"],
  });
```

Maybe you can try again with classic prompt , it will maybe be better .
You have to put the previous input as input variable 
```javascript
 const model3 = new FakeLLM3({});
  const template3 = "Final Answer: {input1} {input2} {input3}.";
  const prompt3 = new PromptTemplate({
    template: template3,
    inputVariables: ["input1", "input2", "input3"],
  });
  const chain3 = new LLMChain({ llm: model3, prompt: prompt3 });
```

**Comment by borel** [comment]
btw it's not really a bug as 'review' is not the result of one of the chain but the initial input . 
i will propose a MR . we will see what happen

**Comment by borel** [comment]
It has been fixed in new version https://github[.com/hwchase17/langchainjs/releases/tag/0.0.92](url) thx to #1508

**Comment by borel** [comment]
you can close please

**Comment by nigel-daniels** [comment]
Hi @borel I appreciate you looking into this, I've updated langchainjs to the latest version 0.0.92 and reran the code with the following results:

1. The original code with `ChatPromptTemplate` fails with the same error, I assumed this should work as following the examples in the lesson that's what they used, but I'm happy to file that as a feature request if that's the suggested route.

2. I update the code to the following:
```
import { ChatOpenAI } from 'langchain/chat_models/openai';
import { PromptTemplate } from 'langchain/prompts';
import { LLMChain, SequentialChain } from 'langchain/chains';

// Note: here we use a higher temp for more creativity
let llm = new ChatOpenAI({temperature: 0.9});

let template4 = 'Translate the following review to english:\n\n{review}';
let promptTemplate4 = new PromptTemplate({template: template4, inputVariables: ['review']});
let llmChain4 = new LLMChain({llm: llm, prompt: promptTemplate4, outputKey: 'english_review'});

let template5 = 'Can you summarize the following review in 1 sentence:\n\n{english_review}';
let promptTemplate5 = new PromptTemplate({template: template5, inputVariables: ['english_review']});
let llmChain5 = new LLMChain({llm: llm, prompt: promptTemplate5, outputKey: 'summary'});

let template6 = 'What language is the following review:\n\n{review}';
let promptTemplate6 = new PromptTemplate({template: template6, inputVariables: ['review']});
let llmChain6 = new LLMChain({llm: llm, prompt: promptTemplate6, outputKey: 'language'});

let template7 = 'Write a follow up response to the following summary in the ' +
    'specified language:\n\nSummary: {summary}\n\nLanguage: {language}';
let promptTemplate7 = new PromptTemplate({template: template7, inputVariables: ['summary', 'language']});
let llmChain7 = new LLMChain({llm: llm, prompt: promptTemplate7, outputKey: 'english_review', 'summary', 'followup_message'});

let sequentialChain = new SequentialChain({
    chains: [llmChain4, llmChain5, llmChain6, llmChain7],
    inputVariables: ['review'],
    outputVariables:['english_review', 'summary', 'followup_message'],
    verbose: true
});

let review1 = 'Je trouve le goût médiocre. La mousse ne tient pas, c\'est bizarre. J\'achète les mêmes dans le commerce et le goût est bien meilleur... Vieux lot ou contrefaçon !?';

//let response3 = await sequentialChain.call(review1);  // Still fails :(
let response3 = await sequentialChain.run(review1);  // Works but throws error

console.log(response3);
```

In the test file above I still see the error, the results are:

- Using `await sequentialChain.call(review1);` results in the previously seen error of `llmChain6` failing to get the `review` input variable.
- Using `await sequentialChain.run(review1);` results in the chain completing but then at the end it throws an error letting me know `run` only supports a single return key at this time.

**Comment by borel** [comment]
small mistake at the end  and it will work !!

```javascript
let review = {
  review: "Je trouve le goût médiocre. La mousse ne tient pas, c\'est bizarre. J\'achète les mêmes dans le commerce et le goût est bien meilleur... Vieux lot ou contrefaçon !?",
}
let response3 = await sequentialChain.call(review);
```

**Comment by nigel-daniels** [comment]
My thanks for helping correct this. Once I am through the course and all is converted to JS from the Python examples I will link my repo to the DeepLearning.AI community site. Hopefully having working examples like this will help other users of `langchainjs` :-)
